subsys/bluetooth/controller/hci/hci.c

```diff
@@ -744,6 +751,239 @@ static void read_bd_addr(struct net_buf *buf, struct net_buf **evt)
+	std_codecs_bytes = num_std_codecs *
+				sizeof(struct bt_hci_std_codec_info);
+	/* read vendor-specific codec information */
+	num_vs_codecs = hci_vendor_read_vs_codecs(&vs_codecs);
+	vs_codecs_bytes = num_vs_codecs *
+				sizeof(struct bt_hci_vs_codec_info);
+
+	rp = hci_cmd_complete(evt, sizeof(*rp) +
+			      std_codecs_bytes +
+			      vs_codecs_bytes);
+	rp->status = 0x00;
+	ptr = &rp->status + sizeof(rp->status);
+
+	/* copy standard codec information */
+	*ptr++ = num_std_codecs;
+	for (i = 0; i < num_std_codecs; i++) {
+		struct bt_hci_std_codec_info codec;
+
+		codec.codec_id = std_codecs[i].codec_id;
+		memcpy(ptr, &codec, sizeof(codec));
+		ptr += sizeof(codec);
+	}
+
+	/* copy vendor specific codecinformation  */
+	*ptr++ = num_vs_codecs;
+	for (i = 0; i < num_std_codecs; i++) {
+		struct bt_hci_vs_codec_info codec;
+
+		codec.company_id = sys_cpu_to_le16(vs_codecs[i].company_id);
+		codec.codec_id = sys_cpu_to_le16(vs_codecs[i].codec_id);
+		memcpy(ptr, &codec, sizeof(codec));
+		ptr += sizeof(codec);
+	}
+}
+
+static void read_codecs_v2(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_rp_read_codecs_v2 *rp;
+	const struct bt_hci_std_codec_info_v2 *std_codecs;
+	const struct bt_hci_vs_codec_info_v2 *vs_codecs;
+	size_t std_codecs_bytes;
+	size_t vs_codecs_bytes;
+	uint8_t num_std_codecs;
+	uint8_t num_vs_codecs;
+	uint8_t *ptr;
+	uint8_t i;
+
+	/* read standard codec information */
+	num_std_codecs = hci_vendor_read_std_codecs(&std_codecs);
+	std_codecs_bytes = num_std_codecs *
+				sizeof(struct bt_hci_std_codec_info_v2);
+	/* read vendor-specific codec information */
+	num_vs_codecs = hci_vendor_read_vs_codecs(&vs_codecs);
+	vs_codecs_bytes = num_vs_codecs *
+				sizeof(struct bt_hci_vs_codec_info_v2);
+
+	/* allocate response packet */
+	rp = hci_cmd_complete(evt, sizeof(*rp) +
+			      std_codecs_bytes +
+			      vs_codecs_bytes);
+	rp->status = 0x00;
+	ptr = (uint8_t *)&rp->status + sizeof(rp->status);
+
+	/* copy standard codec information */
+	*ptr++ = num_std_codecs;
+	for (i = 0; i < num_std_codecs; i++) {
+		struct bt_hci_std_codec_info_v2 codec;
+
+		codec.codec_id = std_codecs[i].codec_id;
+		codec.transport = std_codecs[i].transport;
+		memcpy(ptr, &codec, sizeof(codec));
+		ptr += sizeof(codec);
+	}
+
+	/* copy vendor specific codec information  */
+	*ptr++ = num_vs_codecs;
+	for (i = 0; i < num_std_codecs; i++) {
+		struct bt_hci_vs_codec_info_v2 codec;
+
+		codec.company_id = sys_cpu_to_le16(vs_codecs[i].company_id);
+		codec.codec_id = sys_cpu_to_le16(vs_codecs[i].codec_id);
+		codec.transport = vs_codecs[i].transport;
+		memcpy(ptr, &codec, sizeof(codec));
+		ptr += sizeof(codec);
+	}
+}
+
+uint8_t __weak hci_vendor_read_codec_capabs(uint8_t coding_format,
+					    uint16_t company_id,
+					    uint16_t vs_codec_id,
+					    uint8_t ltt,
+					    uint8_t dir,
+					    uint8_t *num_capabs,
+					    size_t *capabs_bytes,
+					    const uint8_t **capabs)
```

Comment:
```suggestion
uint8_t __weak hci_vendor_read_codec_capabilities(uint8_t coding_format,
					    uint16_t company_id,
					    uint16_t vs_codec_id,
					    uint8_t transport_type,
					    uint8_t direction,
					    uint8_t *num_codec_capabilities,
					    size_t *capability_length,
					    const uint8_t **capability_data)
```

Reply:
I agree to extenting `capabs` to `capabilites`, but I do not think `capability_length`/`capability_data` would be good names here. The intention is that the function returns the whole array of variable-length arrays in `capabs_bytes`/`capabs`, but `capability_length`/`capability_data` would insinuate that it returns a single variable-length array.